chained/tlsfrag.go

```diff
@@ -0,0 +1,40 @@
+package chained
+
+import (
+	"net"
+	"strconv"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport/tlsfrag"
+	"github.com/getlantern/common/config"
+)
+
+func tlsFragConn(conn net.Conn, pc *config.ProxyConfig) net.Conn {
+	splitHello, index := splitHelloInfo(pc)
+	if splitHello {
+		if _, ok := conn.(*net.TCPConn); !ok {
+			return conn
+		}
+		tlsFragConn, err := tlsfrag.WrapConnFunc(conn.(*net.TCPConn), func(record []byte) (n int) {
+			return index
+		})
+		if err != nil {
+			return conn
+		}
+		return tlsFragConn
+	}
+	return conn
+}
+
+func splitHelloInfo(pc *config.ProxyConfig) (bool, int) {
+	indexStr, ok := pc.PluggableTransportSettings["tlsfrag_split_index"]
```

Comment:
Could you implement this in a way that would support more complex future configurations?  The two I currently have in mind are (1) random within a range and (2) splitting the SNI.

My suggestion would be to rename the option `tlsfrag` and expect a value of the form `frag mechanism, { mechanism-specific config }`.  Example values would be `index:7`, `sni` (randomly within the SNI), `sni:3` (byte 3 of the SNI), `random` (randomly anywhere), `random:5,100` (random within bytes 5 to 100).

As-is, it would be a little awkward to add one of these alternative fragmentation mechanisms in a backwards-compatible way.  For example, if we add a new option `tlsfrag_sni`, what does the client do when both that and `tlsfrag_split_index` are specified?

Reply:
I feel like you have a clearer picture as to what that should look like than I do. You want to just run with that and implement it on top of this PR?

Reply:
No problem!

859f05b0cd6a0d11ae48bd1d4a554bbf65317dfc